### DISCUSSION TITLE
Clicking "Back" or pressing "ESC" when adding mods in multiplayer should save changes

### DISCUSSION THREAD

**Discussion by Flamiii** [ppy/osu#25574] [Q&A]
I've seen several people in osu!(lazer) multiplayer trying to add mods or freemods to a map for the first time and they add them and then they either click the "Back" button or press "ESC" and the changes that they made don't save. It can take them a while to realize what they're doing wrong and when they realize that it's because they're not clicking on the osu! cookie or the map itself on song select, pretty much all of them agree that it is pretty unintuitive behavior and it doesn't make much sense.

https://github.com/ppy/osu/assets/74088315/c21a3845-cac7-4e76-bfe8-cc4aae866252

**Comment by peppy** [comment]
I can agree with this. Have also seen it happen multiple times.

Either that, or the mod selection should be moved out of the song select screen. This may be the better path to go, because another common request is being able to set free mods across *all* beatmaps, so that when people add items the free mods are automatically applied (and decided by the host).

**Reply by marioman63** [reply]
This would be a great feature, as long as it doesn't mean removing per map mods. I love having per map mods for special maps that require certain settings, like aspire (i.e., enforcing no HR). maybe a toggle in the lobby options to allow both per map and global mods?
